### PR TITLE
Add a simple Slurm + MPI hello world example

### PR DESCRIPTION
**Pull request by ajdecon**
Getting the examples section started with a very simple MPI example, demonstrated to work in a virtual Slurm DeepOps cluster.

While we will want to have some more impressive examples, e.g. Deep Learning training and all that jazz, we should also have some simple and clear examples that are easy for new users to build on and get running. This example is about as simple as you can get while still being an MPI, but also provides a Slurm usage example.

Next one I'm working on is an equivalent "hello, world" Flask service running in k8s, but running into fun with the registry which is delaying that. So might as well open a PR for this on its own. :)

### CODE REVIEW

**Comment by ajdecon** [comment]
@dholt : Fixed to use the correct path to the slurm deployment guide

File: examples/slurm-mpi-hello/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+It also assumes the presence of a shared NFS filesystem on all nodes in the cluster, which is given the path `/shared` in this example.
+If you haven't configured a Slurm cluster yet, see the [Slurm guide](/docs/slurm-cluster.md) for information on building a GPU-enabled Slurm cluster.
+
+1. **Install the OpenMPI packages:**
+    On many clusters, MPI libraries and tools are built from source to take advantage of the specific cluster hardware.
+    In this example, we'll just use the OpenMPI provided by Ubuntu to demonstrate running an MPI job.
+    To install OpenMPI, run the following `bootstrap-mpi.yml` script using Ansible.
+    If you only want to install on a subset of nodes, use the `-l ${HOST_GROUP}` argument to restrict where this playbook is run.
+    ```
+    $ ansible-playbook -i ${INVENTORY_FILE} [-l ${HOST_GROUP}] bootstrap-mpi.yml
```
Comment (dholt) [inline examples/slurm-mpi-hello/README.md]: I was referring to these paths, i.e. `bootstrap-mpi.yml` here becomes `examples/slurm-mpi-hello/bootstrap-mpi.yml`

File: examples/slurm-mpi-hello/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    ```
+1. Upload the source code and job script to the shared filesystem on your login node:
+    ```
+    $ scp mpi-hello.c login:/shared/
```
Comment (dholt) [inline examples/slurm-mpi-hello/README.md]: Same here, `mpi-hello.c` becomes `examples/slurm-mpi-hello/mpi-hello.c`

**Comment by ajdecon** [comment]
Argh, I screwed up this PR due to an unintentional rebase. Closing this, will open a new PR that's clean.

**Pull request closed without merging**